### PR TITLE
Addition of solutions to 07-control-flow, gapminder challenges

### PR DESCRIPTION
**Pull request by lachlandeer**
In this pull request I add solutions to the gapminder challenges in the 07-control-flow lesson. We use a for loop and if-else statements to loop over each continent or country to calculate life expectancy, and print out statments about the life expectancy (and plot a graph).

I use unique and subset commands, which the students should have seen so far in the R sessions.  I also use grep and cat which should be familar from the Unix Shell.

I only "go in" two deep, one loop, one if-else, because we want to keep the code easy to read, which becomes hard if there are three or more levels of indentation and consequent code

PS. This pull request is part of Instructor Training

### CODE REVIEW

**Comment by naupaka** [comment]
Hi @lachlandeer thanks for this PR, and for addressing one of our open issues! Addresses #135. I don't have the time to go through this immediately but should get to it in the next few days. From first glance looks on target to me.

File: _episodes_rmd/07-control-flow.Rmd [hunk 1]
```diff
@@ -329,14 +329,82 @@ output_vector2
...
+> > We can do that as follows:
+> >
+> > 1. Loop over each of the unique values of 'continent'
+> > 2. For each value of continent, create a temporary variable storing the life exepectancy for that subset,
+> > 3. Return the calculated life expectancy to the user by printing the output:
+> >
+> > ```{r 07-chall-03-sol-b, eval=FALSE}
+> > for( iContinent in unique(gapminder$continent) ){
+> >    tmp_ <- mean(subset(gapminder, continent==iContinent)$lifeExp)
+> >    cat("Average Life Expectancy in", iContinent, "is", tmp, "\n")
```
Comment (naupaka) [inline _episodes_rmd/07-control-flow.Rmd]: variables `tmp` vs `tmp_` are inconsistent

Comment (naupaka) [inline _episodes_rmd/07-control-flow.Rmd]: Also, perhaps good to remove `tmp` after done with the `for()` loop?

File: _episodes_rmd/07-control-flow.Rmd [hunk 2]
```diff
@@ -345,4 +413,44 @@ output_vector2
...
+> > for( iCountry in candidateCountries){
+> >     tmp_lifeExp <- mean(subset(gapminder, country==iCountry)$lifeExp)
+> >     
+> >     if(tmp_lifeExp < thresholdValue){
+> >         cat("Average Life Expectancy in", iCountry, "is less than", thresholdValue, "plotting life expectancy graph... \n")
+> >         
+> >         with(subset(gapminder, country==iCountry),
+> >                 plot(year,lifeExp,
+> >                      type="o",
+> >                      main = paste("Life Expectancy in", iCountry, "over time", sep= " "),
```
Comment (naupaka) [inline _episodes_rmd/07-control-flow.Rmd]: paste has `sep = " "` as the default so no need to include it here.

**Comment by lachlandeer** [comment]
The new commit should have resolved all these issues.

**Comment by naupaka** [comment]
Thanks!

**Pull request merged**